src/pages/project/helpers/projectRoleHelpers.ts

```diff
@@ -0,0 +1,102 @@
+import { ProjectContributor, ProjectContributorRole, ProjectContributorType } from '../../../types/project.types';
+
+export const isProjectManagerRole = (role: ProjectContributorRole) => {
+  return role.type === 'ProjectManager';
+};
+
+export const isNonProjectManagerRole = (role: ProjectContributorRole) => {
+  return role.type !== 'ProjectManager';
+};
+
+export const findProjectManagerRoleIndex = (contributor: ProjectContributor) => {
+  return contributor.roles.findIndex((role) => isProjectManagerRole(role));
+};
+
+export const findProjectManagerRole = (contributor: ProjectContributor) => {
+  const projectManagerRoleIndex = findProjectManagerRoleIndex(contributor);
+
+  return projectManagerRoleIndex > -1 ? contributor.roles[projectManagerRoleIndex] : undefined;
+};
+
+export const replaceRolesOnContributor = (
+  contributors: ProjectContributor[],
+  contributorIndex: number,
+  newRoles: ProjectContributorRole[]
+) => {
+  const newContributors = [...contributors];
+  const newContributor = { ...contributors[contributorIndex] };
+  newContributor.roles = newRoles;
+  newContributors[contributorIndex] = newContributor;
+  return newContributors;
+};
+
+export const deleteRoleFromContributor = (
+  contributors: ProjectContributor[],
+  contributorIndex: number,
+  roleIndex: number
+) => {
+  const newRoles = [...contributors[contributorIndex].roles];
+  newRoles.splice(roleIndex, 1);
+  return replaceRolesOnContributor(contributors, contributorIndex, newRoles);
+};
+
+export const deleteProjectManagerRoleFromContributor = (contributors: ProjectContributor[]) => {
+  const projectManagerIndex = contributors.findIndex((c) => c.roles.some((role) => isProjectManagerRole(role)));
+  const projectManager = contributors[projectManagerIndex];
+  const projectManagerRoleIndex = projectManager?.roles.findIndex((role) => isProjectManagerRole(role));
+
+  return projectManagerRoleIndex > -1
+    ? deleteRoleFromContributor(contributors, projectManagerIndex, projectManagerRoleIndex)
+    : contributors;
+};
+
+const rolesHaveEmptyAffiliation = (roles: ProjectContributorRole[]) => {
+  return roles.some((role) => role.affiliation === undefined);
+};
+
+export const getRelevantContributorRoles = (contributor: ProjectContributor, isProjectManager: boolean) => {
+  return contributor.roles.filter((role) =>
+    isProjectManager ? isProjectManagerRole(role) : isNonProjectManagerRole(role)
+  );
+};
+
+export const contributorHasEmptyAffiliation = (roles: ProjectContributorRole[]) => {
+  return rolesHaveEmptyAffiliation(roles);
+};
+
+export const checkIfSameAffiliationOnSameRoleType = (
+  newAffiliationId: string,
+  role: ProjectContributorRole,
+  isProjectManager: boolean
+) => {
+  const isSameRoleType =
+    (isProjectManager && isProjectManagerRole(role)) || (!isProjectManager && isNonProjectManagerRole(role));
```

Comment:
her er det kanskje også naturlig å bruke rollenavn direkte, både for readability og til forberedelse til nye roller i fremtiden?

```suggestion
  role2: ProjectContributorRole
) => {
  const isSameRoleType = role === role2;
```

Reply:
Samme svar som over. Ta gjerne kontakt om det er noe jeg ikke har tenkt på her, for jeg ser for meg en boolean siden vi har to lister (project manager / ikke project manager) og at det blir mer styr å forholde seg til flere typer project manager ELLER project participant ELLER LocalProjectManager etc. men det er fort noe jeg ikke har tenkt på.